lib/services/azuresqldb/cmd-provision.js

```diff
@@ -0,0 +1,307 @@
+/* jshint camelcase: false */
+/* jshint newcap: false */
+
+var _ = require('underscore');
+var async = require('async');
+var HttpStatus = require('http-status-codes');
+var Config = require('./service');
+var util = require('util');
+var validCollations = require('./validCollations.json');
+
+var sqldbProvision = function (log, params) {
+
+    var instanceId = params.instance_id;
+    var reqParams = params.parameters || {};
+
+    var resourceGroupName = reqParams.resourceGroup || '';
+    var sqldbName = reqParams.sqldbName || '';
+    var sqlServerName = reqParams.sqlServerName || '';
+
+    log.info(util.format('sqldb cmd-provision: resourceGroupName: %s, sqldbName: %s, sqlServerName: %s', resourceGroupName, sqldbName, sqlServerName));
+
+    var location = '';
+    var parametersDefined = !(_.isNull(reqParams.parameters) || _.isUndefined(reqParams.parameters));
+    if (parametersDefined) {
+        location = reqParams.parameters.location || '';
+    }
+
+    var firewallRuleName = '';
+    var firewallRuleStartIp = '';
+    var firewallRuleEndIp = '';
+    if (!_.isUndefined(reqParams.allowSqlServerFirewallRule)) {
+        firewallRuleName = reqParams.allowSqlServerFirewallRule.ruleName;
+        firewallRuleStartIp = reqParams.allowSqlServerFirewallRule.startIpAddress;
+        firewallRuleEndIp = reqParams.allowSqlServerFirewallRule.endIpAddress;
+    }
```

Comment:
I think, `allowSqlServerFirewallRule` should be part of `sqlServerParameters`.
@sebastus